### DISCUSSION TITLE
only on for 15 seconds

### DISCUSSION THREAD

**Discussion by jnsw** [arendst/Tasmota#10503] [Q&A] [answered]
hello all,

is there a function, where you can set a timer, and every time it is turned on it will start the countdown until it automatically turns off? 

thank you

**Comment by barbudor** [comment]
Different solutions
- use a timer to trigger on and Pulsetime set duration
- use a timer to trigger a rule which use a Ruletimer

**Reply by jnsw** [reply]
the thing is that I use my switch directly with alexa over the integrated hue bridge - that means alexa needs to have such functions, but I can't choose it as a trigger for a routine - couldnt be this a feauture request?

**Comment by jdotbdot** [comment, marked as answer]
PulseTime seems to do this for me. I have a smart plug to turn off my espresso machine after 20 minutes.
The auto turn off works if I turn it on by pressing the button or if I switch it on by MQTT.
The setting is retained in the event of a power cut too.

PulseTime n+100 where n is the number of seconds to stay on. eg PulseTime 130 it will turn off after 30 seconds.

**Reply by jnsw** [reply]
i just added a command for 10 seconds with

`PulseTime 110`

it just worked like you said, thank you very much!

**Reply by jnsw** [reply]
also nice to know that it retains it after power loss

**Reply by jdotbdot** [reply]
Edit - typo
Yes, you only need to send PulseTime 110 once. After that any Power ON will be for 10 seconds.

I only acquired the smart plugs this week, I'm not yet fully convinced that it is a foolproof solution - for instance will it still turn off in the event of WiFi or internet outages. Optimistic though.

**Reply by ascillato** [reply]
Yes, pulsetime is a foolproof solution and don't depends on internet or wifi or mqtt. Pulsetime is an internal function. Will work everytime

**Reply by Frogmore42** [reply]
Tasmota doesn't need the internet, so losing it is not an issue for Tasmota (unless that is where your MQTT server is, but even then, it shoudl be okay for turning it off).  Losing Wi-Fi it will depend on your configuration.  There are several options for what Tasmota does when it loses Wi-Fi. See here: https://tasmota.github.io/docs/Commands/#wi-fi

It looks like it should be no issue with the default configuration.
